packages/codegen-ui-react/lib/forms/form-renderer-helper/event-handler-props.ts

```diff
@@ -54,21 +54,25 @@ import { getOnChangeValidationBlock } from './validation';
 import { buildModelFieldObject } from './model-fields';
 import { isModelDataType, shouldWrapInArrayField } from './render-checkers';
 import { extractModelAndKeys, getMatchEveryModelFieldCallExpression } from './model-values';
+import { COMPOSITE_PRIMARY_KEY_PROP_NAME } from '../../utils/constants';
 
-export const buildMutationBindings = (form: StudioForm, primaryKey?: string) => {
+export const buildMutationBindings = (form: StudioForm, primaryKeys: string[] = []) => {
   const {
     dataType: { dataSourceType, dataTypeName },
     formActionType,
   } = form;
   const elements: BindingElement[] = [];
-  if (dataSourceType === 'DataStore' && primaryKey) {
+  if (dataSourceType === 'DataStore' && primaryKeys && primaryKeys.length) {
```

Comment:
Could just check for `.length` for primaryKeys since it defaults to []

Reply:
yes